.github/workflows/ci.yml

```
@@ -0,0 +1,48 @@
+name: ci
+
+on: [push, pull_request]
```

I'm not sure pull_request fit this workflow.
Reading [this doc](https://help.github.com/en/articles/events-that-trigger-workflows) I think we are updating the coverage badge too often right?

What do you think about to split this in two workflows, one dedicated for push to master that update the coverage and one push that trigger tests? I say push because of [this suggestion](https://help.github.com/en/articles/events-that-trigger-workflows#pull-request-events-for-forked-repositories)

---

I really don't understand what that note is saying. Is it saying that a PR from a forked repository will not run tests on the forked code when the `pull_request` event is received?

---

That does not seem to be the case -- https://github.com/ldapjs/node-ldapjs/pull/553/checks?check_run_id=222806190#step:5:466

The test that link is referencing only exists in the forked repo.

---

> The test that link is referencing only exists in the forked repo.

Thanks, I didn't get that, so the `pull_request` is needed, gotcha 👍

My concern is about this use case: if we ask for a review or add a label the `pull_request` event is triggered and the step `Generate coverage` is executed.
I think the update of the badges could happen only on push on master

---

This might solve it:

```suggestion
on:
  - push
  - pull_request:
    types:
    - synchronized
```

---

@jsumners what does `types` means?

---

Nevermind, nice tip,

---

@jsumners @Eomm check here https://help.github.com/en/articles/events-that-trigger-workflows the `pull_request` is only triggered opened, synchronize, reopened that should be fine.

---

I understand that table now. Man, whomever is writing the documentation for Actions has a very weird and hard to follow writing style.

Yeah, just stick with `pull_request`. No need to declare the `types` that trigger it.

---

Clear, thanks for the check